sycl/source/detail/queue_impl.hpp

```diff
@@ -388,6 +388,19 @@ class queue_impl {
   template <typename Param>
   typename Param::return_type get_backend_info() const;
 
+  /// Begin to execute previously issued commands on this queue
+  /// if they have not been executed yet. Overrides normal batching behaviour.
```

Comment:
We should explicitly indicate here that this is hint only, because this gives a wrong impression that it is guaranteed that all previous commands will start executing.

I believe that there can be situations in SYCL scheduler when command which is submitted to the scheduler is not enqueued to the PI queue right away (if this enqueue is blocked by something, most like this can happen when host tasks are involved), so piQueueFlush will not flush such command.

Since this is a hint only per documentation and this only preliminary implementation, I believe just calling piQueueFlush is fine.